eskip/eskip.go

```diff
@@ -96,6 +96,21 @@ type Route struct {
 	Backend string
 }
 
+type RoutePredicate func(*Route) bool
+
+type RouteList []*Route
+
+// RouteInfo contains a route id, plus the loaded and parsed route or
+// the parse error in case of failure.
+type RouteInfo struct {
```

Comment:
this type definition feels a bit orphan in this package, not sure if this is the right place. I remember we discussed it, but. The reason why it is currently in the etcd package is that in etcd, being a generic storage with an open API, one can store invalid routes or just any data, and depending on the client, it can be useful to report parse errors individually. Currently, this is only used by the eskip tool, that verifies the whether one or more route definitions are valid. The runtime configuration rejects the whole set of routes, in case of any one of them is invalid, because that indicates more serious problems than just a single parse error. But this makes sense only in case of etcd, because as I checked, the recent Innkeeper responses, once we have a valid json response, they will be valid eskip routes. See the `LoadAndParseAll` in innkeeper/client.go of this pull-request, it never sets a parse error. So, I think we should use this type only in the etcd client.